### PR TITLE
Documents service

### PR DESCRIPTION
**Pull request by JosephSC0121**
## Description

Se siguio la estructura planteada anteriormente.
 
Servicio format
-get by id
image

-create
image

## How Has This Been Tested?

postman

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Alejandro-Suarez-Rincon — APPROVED** [review]
Usar Hoppscothc para que todos tengamos acceso a las peticiones.

File: database/schemas/20240526020158_document.up.sql
Comment (CamiloArias12) [inline database/schemas/20240526020158_document.up.sql:1]: Deberia tener un nombre para mostrarlo asi
image

Comment (JosephSC0121) [reply database/schemas/20240526020158_document.up.sql:1]: no entendi

File: internal/config/config.go [hunk 1]
```diff
@@ -18,7 +18,16 @@ type EnvStruct struct {
...
+type DigitalOceanStruct struct {
+	DigitalOceanKey      string
+	DigitalOceanSecret   string
+	DigitalOceanEndpoint string
+	DigitalOceanRegion   string
+	DigitalOceanBucket   string
+}
+
 var Env *EnvStruct
+var DigitalOcean *DigitalOceanStruct
```
Comment (Camilo-Camargo) [inline internal/config/config.go:30]: me parece mejor dejarlo en Env.

File: internal/repository/documents.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	return &dto.GetDocumentRes{
+		Id:          repoRes.DocumentID,
+		DirectoryId: repoRes.DirectoryID,
+		FormatId:    repoRes.FormatID.Int32,
+		FileRute:    repoRes.FileRute,
+		Status:      string(repoRes.Status),
```
Comment (Camilo-Camargo) [inline internal/repository/documents.go:34]: porque string()

File: internal/service/documents/documents.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		Region:           aws.String(cnf.DigitalOcean.DigitalOceanRegion),
+	}
+
+	sess, err := session.NewSession(s3Config)
+
+	if err != nil {
+		return "Error"
+	}
+
+	//TODO: Change this for multipart
```
Comment (Camilo-Camargo) [inline internal/service/documents/documents.go:72]: @CamiloArias12

**Review by Camilo-Camargo — CHANGES_REQUESTED** [review]

File: internal/repository/documents.go (hunk 2 above)
Comment (JosephSC0121) [reply internal/repository/documents.go:34]: pq como es un ENUM sqlc lo sqlc lo toma como tipo STATUS, entonces para obtener el valor string se hace el cast
image

File: internal/config/config.go (hunk 1 above)
Comment (JosephSC0121) [reply internal/config/config.go]: [f365b5a](https://github.com/OwlByTech/optitech-api/pull/24/commits/f365b5a8801008faa32b675b535b2e1e7b95b61c)****

File: database/schemas/20240526020158_document.up.sql
Comment (CamiloArias12) [reply database/schemas/20240526020158_document.up.sql:1]: image
agregar el campo name para mostrarlo

File: internal/dto/document/create.go [hunk 4]
```diff
@@ -1,18 +1,16 @@
 package dto
 
-type CreateDocumentClientReq struct {
-	FormatId      int64  `json:"formatId" validate:"required"`
-	InstitutionId int64  `json:"institutionId" validate:"required"`
-	ClientId      int64  `json:"clientId" validate:"required"`
-	FileRute      string `json:"fileRute" validate:"required"`
-	Status        string `json:"status" validate:"required"`
+type CreateDocumentReq struct {
```
Comment (Camilo-Camargo) [inline internal/dto/document/create.go:3]: Falta agregar nombre de documento

File: internal/service/documents/documents.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		return nil, err
+	}
+
+	// TODO: RETURNS EMPTY JSON
+
+	document := &dto.CreateDocumentRes{
+		Id: repoRes.Id,
+	}
+	return document, err
+
```
Comment (Camilo-Camargo) [inline internal/service/documents/documents.go:54]:
```suggestion
```

**Review by Camilo-Camargo — APPROVED** [review]

**Pull request merged**